### ISSUE TITLE
(Feature request) Add synthetic gradient training per "Decoupled Neural Interfaces using Synthetic Gradients"

### ISSUE THREAD

**Issue by AzizCode92** [tensorflow/tensorflow#22935]
### Related papers
https://arxiv.org/abs/1703.00522
### Related blog : https://deepmind.com/blog/decoupled-neural-networks-using-synthetic-gradients/

**System information**
Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No

**OS Platform and Distribution (e.g., Linux Ubuntu 16.04):**

VERSION="16.04.4 LTS (Xenial Xerus)"
VERSION_ID="16.04"
VERSION_CODENAME=xenial

**Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:**
No

**TensorFlow installed from (source or binary):**
source

**TensorFlow version (use command below):**
tf.VERSION = 1.9.0

**Python version:** python 2.7

**Bazel version (if compiling from source):**
0.11.1

**GCC/Compiler version (if compiling from source):**
c++ (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609

**CUDA/cuDNN version:**
9.1

**GPU model and memory:**
Tesla K80

**Exact command to reproduce:**
( No command)

**Comment by AzizCode92** [comment]
I have updated the template.

**Comment by wt-huang** [comment]
Thanks for bringing this to our attention. Synthetic gradient is a neat approach, we will note this down. You can also implement it yourself using TensorFlow and common python libraries.
